### PR TITLE
[inductor] getting AOT inductor to treat None args correctly

### PR DESCRIPTION
**Pull request by SamGinzburg**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #139114

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

Differential Revision: [D65102228](https://our.internmc.facebook.com/intern/diff/D65102228)

### CODE REVIEW

File: torch/_inductor/ir.py [hunk 1]
```diff
@@ -5414,17 +5414,49 @@ def codegen(self, wrapper):
...
+            if idx in removed_none_args:
+                index_shift += 1
+                continue
+            arg_index = kernel.arg_names.index(kwarg)
+            if arg_index in kernel.constexprs:
+                constexpr_indices.append(idx - index_shift)
+            if arg_index in eq1_indices_set:
+                equal_to_1.append(idx - index_shift)
+
+        triton_meta["configs"][0].equal_to_1 = equal_to_1
```
Comment (aakhundov) [inline torch/_inductor/ir.py:5449]: Do this only `if removed_none_args`?

**Comment by SamGinzburg** [comment]
@SamGinzburg has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D65102228).

**Review by aakhundov — APPROVED** [review]

**Pull request closed without merging**